libs/application-generic/src/usecases/workflow/delete-workflow/delete-workflow.usecase.ts

```diff
@@ -78,6 +84,31 @@ export class DeleteWorkflowUseCase {
         });
       }
 
+      const upsertUserWorkflowPreferencesPromise =
+        this.upsertPreferences.upsertUserWorkflowPreferences(
```

Comment:
This is effectively a preference deletion. Do we need to use the upsert workflow preference use case? The naming is off, as upsert implies create or update. Passing `null` to indicate removal is a dangerous pattern.

I also suggest we add a transaction. See [here](https://github.com/novuhq/novu/blob/5a0efd7f02cb4c579c65439751daefa029cc551e/apps/api/src/app/subscribers/usecases/remove-subscriber/remove-subscriber.usecase.ts#L58).

Reply:
I agree, I have added a new `DeletePreference` use-case to explicitly handle deletion now.

Regarding the transaction, this would necessitate passing a transaction reference across use-cases, which isn't something that we have a pattern for. Do we think adding a transaction now is a requirement, or should we address it separately?

Reply:
Scratch the above, I'm adding a transaction.

Reply:
Transactions are now in place for all the workflow mutation use-case - `CreateWorkflow`, `UpdateWorkflow`, `DeleteWorkflow`.